### PR TITLE
[xm] Support UseHardenendRuntime in code signing

### PR DESCRIPTION
**Pull request by chamons**
- https://github.com/xamarin/xamarin-macios/issues/4288

There is some IDE work needed to full support hardened runtime, but this is enough to allow people to support it if they are willing to use a text editor on their csproj.

### CODE REVIEW

File: msbuild/Xamarin.Mac.Tasks/Xamarin.Mac.Common.targets [hunk 1]
```diff
@@ -249,6 +249,7 @@ Copyright (C) 2014 Xamarin. All rights reserved.
 			SigningKey="$(_CodeSigningKey)"
 			ExtraArgs="$(CodesignExtraArgs)"
 			IsAppExtension="$(IsAppExtension)"
+			UseHardenendRuntime="$(UseHardenendRuntime)"
```
Comment (rolfbjarne) [inline msbuild/Xamarin.Mac.Tasks/Xamarin.Mac.Common.targets]: Typo: `Hardenend`-> `Hardened`.

Comment (chamons) [reply msbuild/Xamarin.Mac.Tasks/Xamarin.Mac.Common.targets]: 🤦‍♂️

**Comment by chamons** [comment]
🤔 
```
10:31:42.0558860   /Users/builder/jenkins/workspace/xamarin-macios-pr-builder/_mac-build/Library/Frameworks/Mono.framework/External/xbuild/Xamarin/Mac/Xamarin.Mac.Common.targets(446,3): error : Could not find any available provisioning profiles for OS X. [/Users/builder/jenkins/workspace/xamarin-macios-pr-builder/tests/xharness/tmp-test-dir/mmptest/bin/Debug/tmp-test-dir/Xamarin.MMP.Tests.MMPTests.RunMMPTest84/UnifiedExample.csproj]
```
Build Bot w\ no certificates?

**Comment by chamons** [comment]
Ok, this passes locally so I need to figure out what's going on (or just convert this to a manual test).

**Comment by rolfbjarne** [comment]
build

**Comment by chamons** [comment]
I believe the failure is https://github.com/xamarin/maccore/issues/1412

**Comment by chamons** [comment]
@monojenkins backport d16-0

**Pull request merged**